backend/src/controllers/AccountController.ts

```diff
@@ -0,0 +1,166 @@
+import { Request, Response } from "express";
+import { supabase } from "../services/supabase";
+import ErrorHandling from "../util/ErrorHandling";
+import { z } from "zod";
+
+const AccountRef = supabase.from("account");
+
+const insertBodySchema = z.object({
+  email: z.string().email().min(1),
+  phonenumber: z.string().min(1),
+  password: z.string().min(1),
+  accounttypeid: z.number().int(),
+});
```

Comment:
isso mesmoo, só deixei uma sugestão de nomenclatura pra ficar mas claro
```suggestion
const accountSchema = z.object({
  email: z.string().email().min(1),
  phonenumber: z.string().min(1),
  password: z.string().min(1),
  accounttypeid: z.number().int(),
});
```


e uma dica pro futuro é que uma forma de organizar bacana também é criar um arquivo separado de schema pro controller, que ae inclusive pode servir pra tipagem dos models

exemplo, ao inves de definir o schema e a tipagem separada quando for precisar usar em algum lugar é só fazer tipo: type Account = typeof AccountSchema 🚀